### ISSUE TITLE
False positive on "Cannot move" inspection with Copy trait bound

### ISSUE THREAD

**Issue by xpe** [intellij-rust/intellij-rust#3093]
## Environment

* Intellij-Rust plugin version: v0.2.0.2111-183
* Rust toolchain version: rustup 1.15.0 (f0a3d9a84 2018-11-27)
* cargo 1.32.0-nightly (b3d0b2e54 2018-11-15)
* IDE name and version: CLion 2018.3 Build #CL-183.4284.156
* Operating system: macOS 10.14.1 (18B75)

## Problem description

I get an incorrect inspection on the following code, saying "Cannot move" (borrow checking).

<img width="663" alt="rust-fn-1" src="https://user-images.githubusercontent.com/4576329/49248286-37d02300-f3ce-11e8-9cd0-c15151625fc9.png">

<img width="649" alt="rust-fn-2" src="https://user-images.githubusercontent.com/4576329/49248310-44547b80-f3ce-11e8-90d3-de3edb8b6257.png">

## Steps to reproduce

The function from above:

```rust
/// Returns a tuple `(index, v)` such that `v` is the maximum.
pub fn argmax<V>(vs: &[V]) -> (usize, V) where V: Copy + PartialOrd {
    assert_ne!(vs.len(), 0);
    let mut max_i = 0;
    let mut max_v = vs[0];
    for (i, &v) in vs.iter().enumerate().skip(1) {
        if max_v < v {
            max_i = i;
            max_v = v;
        }
    }
    (max_i, max_v)
}
```

Cargo.toml:

```toml
[package]
name = "foo"
version = "0.1.0"
authors = ["First Last <person@place.org>"]
edition = "2018"

[dependencies]
```

**Comment by xpe** [comment]
Below, I change the trait bound syntax slightly, and you'll see the warning still exists:

<img width="597" alt="rust-fn-3" src="https://user-images.githubusercontent.com/4576329/49248590-f1c78f00-f3ce-11e8-95c6-39eec112d18d.png">

**Comment by xpe** [comment]
Here is an even simpler way to reproduce the false positive:

<img width="354" alt="rust-fn-4" src="https://user-images.githubusercontent.com/4576329/49248746-4bc85480-f3cf-11e8-89af-f6673c2abdd1.png">

**Comment by xpe** [comment]
Possibly related issues include:
* #3062

**Comment by ortem** [comment]
Thank you for the detailed report!
It is really related to #3062. Moreover, the corresponding fix has added this bug 😄.

**Comment by alexander-irbis** [comment]
One more example
```rust
fn copy<X: Copy>(x: X) -> (X, X) {
    (x, x)
}
```
![screenshot_20181201_013431](https://user-images.githubusercontent.com/1448323/49320309-9b12a000-f509-11e8-914d-1ef4d8cf4b1c.png)

**Comment by alexander-irbis** [comment]
And yet another one
```rust
#[derive(Copy)]
struct X {}

fn copy(x: X) -> (X, X) {
    (x, x)
}
```

**Comment by ortem** [comment]
@alexander-irbis your first example is related to this bug (and fixed in #3094). But the second is not about type parameters, and I even can't reproduce it. Do you really have the inspection error in the second case?

**Comment by alexander-irbis** [comment]
@ortem I initially came with this bug, but found this issue and decided this is the same.
![screenshot_20181201_184030](https://user-images.githubusercontent.com/1448323/49330531-b5e02580-f598-11e8-9c84-29a5835a1bf1.png)

**Comment by ortem** [comment]
@alexander-irbis Could you please attach your environment configuration?
